### PR TITLE
*: Support character string literal

### PR DESCRIPTION
**Pull request by shenli**
See: https://dev.mysql.com/doc/refman/5.7/en/charset-literal.html
Done: SELECT _latin1'string';
TODO:   SELECT _latin1'string' COLLATE latin1_danish_ci;

### CODE REVIEW

File: stmt/stmts/select_test.go [hunk 1]
```diff
@@ -33,6 +33,18 @@ func (s *testStmtSuite) TestSelectWithoutFrom(c *C) {
 
 	rows.Close()
 	mustCommit(c, tx)
+
+	tx = mustBegin(c, s.testDB)
+	rows, err = tx.Query(`select _utf8"string";`)
+	c.Assert(err, IsNil)
+
+	for rows.Next() {
```
Comment (siddontang) [inline stmt/stmts/select_test.go]: you can use matchRows ti simplify this.

File: util/charset/charset.go [hunk 2]
```diff
@@ -118,11 +120,20 @@ func ValidCharsetAndCollation(cs string, co string) bool {
 func GetDefaultCollation(charset string) (string, error) {
 	c, ok := charsets[charset]
 	if !ok {
-		return "", fmt.Errorf("Unkown charset %s", charset)
+		return "", errors.Errorf("Unkown charset %s", charset)
 	}
 	return c.DefaultCollation.Name, nil
 }
 
+// GetCharsetInfo return charset and collation for cs as name.
```
Comment (zimulala) [inline util/charset/charset.go]: return -> returns?

File: parser/parser.y [hunk 3]
```diff
@@ -1876,11 +1877,26 @@ Literal:
...
+	}
+|	"UNDERSCORE_CHARSET" stringLit
+	{
+		// See: https://dev.mysql.com/doc/refman/5.7/en/charset-literal.html
+		tp := types.NewFieldType(mysql.TypeString)
+		tp.Charset = $1.(string)
+		co, err := charset.GetDefaultCollation(tp.Charset)
+		if err != nil {
+			l := yylex.(*lexer)
+			l.err(fmt.Sprintf("Get collation error for charset: %s", tp.Charset))
```
Comment (zimulala) [inline parser/parser.y]: need `return 1`?

**Comment by shenli** [comment]
PTAL

**Comment by zimulala** [comment]
LGTM

File: parser/parser.y (hunk 3 above)
Comment (qiuyesuifeng) [inline parser/parser.y]: l.errf?

Comment (shenli) [inline parser/parser.y]: PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**